include/vapor/GeoImageTMS.h

```diff
@@ -9,13 +9,78 @@
 #include <xtiffio.h>
 #include <geotiff.h>
 #endif
+#include <sstream>
+#include <fstream>
+#include <sys/stat.h>
 #include <vapor/MyBase.h>
 #include <vapor/UDUnitsClass.h>
 #include "GeoTileMercator.h"
 #include "GeoImage.h"
 
 namespace VAPoR {
 
+static inline bool IsTMSFile( 
+    std::string path 
+) {
+    if ( path.rfind(".tms", path.size()-4) != string::npos ) {
+        ifstream in;
+        in.open( path.c_str() );
+        if ( ! in ) {
+            GeoImage::SetErrMsg("fopen(%s) : %M", path.c_str());
+            return false ;
+        }
+        in.close();
+    }
+
+    else {
+        return false;
+    }
+
+    return true;
+}
+
+// Get the file path of a single tile from the TMS database
+// with the give lod and tile coordinates
+//
+static inline std::string TilePath(
+    string dir, size_t tileX, size_t tileY, int lod
+) {
+    size_t tmsTileY = tileY;
+
+    ostringstream oss;
+    oss << dir;
+    oss << "/";
+    oss << lod;
+    oss << "/";
+    oss << tileX;
+    oss << "/";
+    oss << tmsTileY;
+
+    string base = oss.str();
+
+    string path = base + ".tif";
+
+    struct stat statbuf;
+    if (stat(path.c_str(), &statbuf) == 0)  return(path);
+
+    path = base + ".tiff";
+
+    if (stat(path.c_str(), &statbuf) == 0) return (path);
+
+    // Tile does not exist
+    //
+    return("");
+}
+
+static inline int GetAvailableTMSLODs( std::string dir ) {
```

Comment:
Rather than having this function operate on a directory how about having it operate on the .tms file? Otherwise the caller has to unnecessarily know the details about how TMS data bases are structured.

Reply:
The logic here is legacy code, which was just moved from the .cpp file into a helper function in the .h file.

I agree that this is awkward.  I'm always trying to minimize the changes that come with my PRs, because more changes require more testing and more attention from reviewers.

I made a judgment call that refactoring this logic was outside of the scope of this PR.  But we can fix it in this PR too.  Let me know if this is something you'd like me to do.

Reply:
Please fix

Reply:
Fixed.